src/compiler/checker.ts

```diff
@@ -5341,37 +5341,53 @@ namespace ts {
                 if (isArrayType(type)) {
                     return createArrayType(getWidenedType((<TypeReference>type).typeArguments[0]));
                 }
+                if (isTupleType(type)) {
+                    return createTupleType(map(type.elementTypes, getWidenedType));
+                }
             }
             return type;
         }
 
-        function reportWideningErrorsInType(type: Type): boolean {
+        function reportWideningErrorsInType(type: Type, expression: Expression): boolean {

```

Comment:
`expression` is the expression to report the error on. I can rename it `expressionForError`.

What do you mean by `the context for which you're feeding in type`? And what are you asking when I will need to do?